homeassistant/components/influxdb/config_flow.py

```diff
@@ -0,0 +1,62 @@
+"""Config flow for InfluxDB integration."""
+
+import logging
+from typing import Any
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+
+from . import DOMAIN, get_influx_connection
+from .const import CONF_API_VERSION, CONF_BUCKET, CONF_DB_NAME, DEFAULT_API_VERSION
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _validate_influxdb_connection(
+    hass: HomeAssistant, data: dict[str, Any]
+) -> dict[str, str] | None:
+    """Validate connection to influxdb."""
+
+    try:
+        influx = await hass.async_add_executor_job(get_influx_connection, data, True)
+    except ConnectionError as ex:
+        _LOGGER.error(ex)
+        return {"base": "cannot_connect"}
+    except Exception:
+        _LOGGER.exception("Unknown error")
+        return {"base": "unknown"}
+
+    influx.close()
+
+    return None
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
```

Comment:
Should we also add the user step? This way we can properly deprecate the YAML config

Reply:
In a follow-up PR :)